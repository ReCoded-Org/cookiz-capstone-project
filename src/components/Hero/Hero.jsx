import { useRouter } from "next/router";
import { useTranslation } from "next-i18next";
import { useEffect, useState } from "react";

function Hero() {
    const [search, setSearch] = useState("");
    const { t } = useTranslation("hero");

    useEffect(() => {}, [search]);

    // intializing router
    const router = useRouter();

    // redirect handler
    const handleRedirect = (event, PATH) => {
        event.preventDefault();
        router.push(PATH);
    };

    return (
        <section className='hero-container mb-[10rem] flex h-screen items-center bg-herobg bg-cover pl-[10rem]'>
            <div className='wrapper flex h-[30vw] w-auto max-w-[1920px] flex-col justify-center p-[2rem]'>
                <div className='hero-title mb-[3rem] mt-[-10rem]'>
                    <h1 className='max-w-[35vw] text-[5rem] font-bold leading-[1] text-orange-700'>
                        {t("title")}
                    </h1>
                </div>
                <div className='hero-search-section box-border'>
                    <div className='search-container mb-[2rem] flex flex-wrap items-center'>
                        {/* search input */}
                        <div className='search-box mr-3'>
                            <input
                                className='pt-6 pb-6 text-[24px] sm:w-96'
                                type='text'
                                placeholder={t("placeholder")}
                                onChange={(e) => setSearch(e.target.value)}
                                value={search}
                            />
                        </div>
                        <div className='search-btn'>
                            <button
                                onClick={(e) => {
                                    if (search != "") {
                                        handleRedirect(e, "/LandingMeal");
                                    } else {
                                        console.log(
                                            "Enter a food name or location"
                                        );
                                    }
                                }}
                                className='box-border bg-yellow-600 px-10 py-[23px] text-[24px] font-medium text-[white] hover:bg-yellow-700'
                            >
                                {t("search-button")}
                            </button>
                        </div>
                    </div>
                    <h6 className='pl-[10px] text-[0.9vw] text-gray-800'>
                        {t("subtitle")}
                    </h6>
                </div>
            </div>
        </section>
    );
}

export default Hero;
