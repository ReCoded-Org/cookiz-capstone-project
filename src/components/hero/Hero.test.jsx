import React from "react";
import renderer from "react-test-renderer";

import Hero from "@/components/hero/Hero";
// eslint-disable-next-line
it("Hero Component", () => {
    const tree = renderer.create(<Hero />).toJSON();
    // eslint-disable-next-line no-undef
    expect(tree).toMatchSnapshot();
});
