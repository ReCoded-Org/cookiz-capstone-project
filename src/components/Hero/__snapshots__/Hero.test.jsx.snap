// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Component 1`] = `
<section
  className="hero-container mb-[10rem] flex h-screen items-center bg-herobg bg-cover pl-[10rem]"
>
  <div
    className="wrapper flex h-[30vw] w-auto max-w-[1920px] flex-col justify-center p-[2rem]"
  >
    <div
      className="hero-title mb-[3rem] mt-[-10rem]"
    >
      <h1
        className="max-w-[35vw] text-[5rem] font-bold leading-[1] text-orange-700"
      >
        Fast delivery, delicious and cheaper
      </h1>
    </div>
    <div
      className="hero-search-section box-border"
    >
      <div
        className="search-container mb-[2rem] flex items-center"
      >
        <div
          className="search-box mr-3"
        >
          <input
            className="w-[800px] pt-6 pb-6 text-[24px]"
            onChange={[Function]}
            placeholder="Enter delivery address"
            type="text"
            value=""
          />
        </div>
        <div
          className="search-btn"
        >
          <button
            className="box-border bg-yellow-600 px-10 py-[23px] text-[24px] font-medium text-[white] hover:bg-yellow-700"
          >
            Find
          </button>
        </div>
      </div>
      <h6
        className="pl-[10px] text-[0.9vw] text-gray-800"
      >
        Find your favorite home made food and order in seconds
      </h6>
    </div>
  </div>
</section>
`;
